clang/lib/Driver/ToolChains/SPIRV.cpp

```diff
@@ -95,7 +95,19 @@ void SPIRV::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   CmdArgs.push_back("-o");
   CmdArgs.push_back(Output.getFilename());
 
+  // Use of --sycl-link will call the clang-sycl-link-wrapper instead of
+  // the default linker (spirv-link).
+  if (Args.hasArg(options::OPT_sycl_link))
+    Linker = ToolChain.GetProgramPath("clang-sycl-link-wrapper");
```

Comment:
Right, I have a better understanding now than when I wrote https://github.com/asudarsa/llvm-project/pull/1#issuecomment-2402511973

You shouldn't need that flag... if the driver was there. If it was, the condition should be `if (JA.isDeviceOffloading(Action::OFK_SYCL))` rather than this global modifier. IMHO, the driver change shouldn't be part of this patch:
 - there is a better way to do it but can't implement this because it lacks the required sycl offloading driver logic.
 - looking at the test, you don't need this to test `clang-sycl-linker-wrapper`

Reply:
Thanks for the feedback @Naghasan 
Here, the clang driver is called to link device images and we pass --sycl-link (and no -fsycl) to invoke the 'link' stage.
Adding @mdtoguchi for more details here. 

Thanks

Reply:
The use of `JA.isDeviceOffloading()` is useful when we know we are offloading and determining what kind we are offloading to.  From what I understand, the `clang` call from within the wrapper is not to be considered as an offloading enabling invocation, but rather a target specific call to produce a specific type of binary.